### PR TITLE
Reorder

### PR DESCRIPTION
**Pull request by mattgperry**
TODO:

- [x] Test for x axis
- [x] Test custom components rendered correctly

### CODE REVIEW

File: src/components/Reorder/Group.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import * as React from "react"
+import { FunctionComponent, ReactHTML, useEffect, useRef } from "react"
+import { ReorderContext } from "../../context/ReorderContext"
+import { motion } from "../../render/dom/motion"
+import { HTMLMotionProps } from "../../render/html/types"
+import { useConstant } from "../../utils/use-constant"
+import { ItemData, ReorderContextProps } from "./types"
+import { checkReorder } from "./utils/check-reorder"
+
+interface Props<V> {
```
Comment (shuangq) [inline src/components/Reorder/Group.tsx:10]: Why don't we use the `RenderGroupProps` type directly here?

File: dev/examples/Drag-drag-to-reorder.tsx [hunk 2]
```diff
@@ -80,7 +74,7 @@ export const App = () => {
     // sibling.
     const moveItem = (i: number, dragOffset: number) => {
         const targetIndex = findIndex(i, dragOffset, positions)
-        if (targetIndex !== i) setColors(move(colors, i, targetIndex))
+        // if (targetIndex !== i) setColors(move(colors, i, targetIndex))
```
Comment (shuangq) [inline dev/examples/Drag-drag-to-reorder.tsx:77]: Is this still a todo?

File: dev/tests/drag-to-reorder.tsx [hunk 3]
```diff
@@ -1,166 +1,144 @@
...
-        <li
-            style={{
-                padding: 0,
-                height: heights[color],
-                zIndex: isDragging ? 3 : 1,
-            }}
+        <Reorder.Item
+            value={item}
+            id={item}
+            drag
```
[the remark this answers is not in the document]
Comment (mattgperry) [reply dev/tests/drag-to-reorder.tsx:36]: By default `drag` is the axis we can reorder in. By passing `drag` manually we can drag on both axes even when sorting in just one

File: src/components/Reorder/Group.tsx (hunk 1 above)
Comment (mattgperry) [reply src/components/Reorder/Group.tsx:10]: I should have removed `RenderGroupProps`

File: dev/examples/Drag-drag-to-reorder.tsx (hunk 2 above)
Comment (mattgperry) [reply dev/examples/Drag-drag-to-reorder.tsx:77]: I can just delete this file as there's a test

File: api/framer-motion.api.md [hunk 4]
```diff
@@ -1096,13 +1102,17 @@ export interface VisualElementLifecycles {
...
     onUnmount?(): void;
     onUpdate?(latest: ResolvedValues): void;
 }
 
 
+// Warnings were encountered during analysis:
+// 
+// types/components/Reorder/index.d.ts:7:5 - (ae-forgotten-export) The symbol "Group" needs to be exported by the entry point index.d.ts
+// types/components/Reorder/index.d.ts:8:5 - (ae-forgotten-export) The symbol "Item" needs to be exported by the entry point index.d.ts
+
```
[the remark this answers is not in the document]
Comment (mattgperry) [reply api/framer-motion.api.md]: I'm not sure what its problem is as these are exported

**Review by shuangq — APPROVED** [review]

**Pull request merged**